src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -985,6 +986,133 @@ ExitCode ServerRequests::getUserFromSyncDbId(int syncDbId, User &user) {
     return ExitCodeOk;
 }
 
+ExitCode ServerRequests::sendLogToSupport(bool includeArchivedLog, std::function<bool(LogUploadState, int)> progressCallback,
+                                          ExitCause &exitCause) {
+    exitCause = ExitCauseUnknown;
+    ExitCode exitCode = ExitCodeOk;
+    std::function<bool(LogUploadState, int)> safeProgressCallback =
+        progressCallback != nullptr
+            ? std::function<bool(LogUploadState, int)>(
+                  [progressCallback](LogUploadState status, int percent) { return progressCallback(status, percent); })
+            : std::function<bool(LogUploadState, int)>([](LogUploadState, int) { return true; });
+
+    safeProgressCallback(LogUploadState::Archiving, 0);
+
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, std::string(""), found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+        // Do not return here because it is not a critical error, especially in this context where we are trying to send logs
+    }
+
+    SyncPath logUploadTempFolder;
+    IoError ioError = IoErrorSuccess;
+
+    IoHelper::logArchiverDirectoryPath(logUploadTempFolder, ioError);
+    if (ioError != IoErrorSuccess) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in IoHelper::logArchiverDirectoryPath : "
+                                                   << Utility::formatIoError(logUploadTempFolder, ioError).c_str());
+        return ExitCodeSystemError;
+    }
+
+    IoHelper::createDirectory(logUploadTempFolder, ioError);
+    if (ioError == IoErrorDirectoryExists) {  // If the directory already exists, we delete it and recreate it
+        IoHelper::deleteDirectory(logUploadTempFolder, ioError);
+        IoHelper::createDirectory(logUploadTempFolder, ioError);
```

Comment:
It is the desired behavior,  IoHelper::createDirectory (line 1017) fail with IoErrorDirectoryExists, as this just indicates that the previous log upload failed (and had not clean up). To prevent the accumulation of log archives, I delete the directory here. If there is no error in this process, IoHelper::createDirectory (line 1020) will set ioError to IoErrorSuccess; otherwise, it will be caught at line 1023. This should not be a problem for the createDirectory function either, as all IoHelper functions internally reset the ioError at their beginning.

Reply:
Do you have a specific issue in mind that this could lead to?